andela/ah-scorpion
# #159053987 Add functionality for rating articles

**What does this PR do?**
Adds the functionality to view an article's ratings.and to rate an article.

**Description of Task to be completed?**
1. Add a new app called Ratings with its model that reference both the User and Article model
2. Modify ArticleSerializer to have two new fields, **ratingsCount** and **averageRating**.
3. Add a method to enable a user to rate an article with a number of stars.

**How can this be manually tested?**
- Clone the repository from [here](https://github.com/andela/ah-scorpion/tree/ft-Article-Rating-%23159053987)
- Activate your virtual environment and use the `.env-sample` to set the environment variables.
- Use postman to test these endpoints.
  `GET api/v1/articles/<str:slug>/`, the response should look something like this(notice the two new fields):
```source-json
{
  "article": {
    "slug": "how-to-train-your-dragon",
    "title": "How to train your dragon",
    "description": "Ever wonder how?",
    "body": "It takes a Jacobian",
    "tagList": ["dragons", "training"],
    "createdAt": "2016-02-18T03:22:56.637Z",
    "updatedAt": "2016-02-18T03:48:35.824Z",
    "favorited": false,
    "favoritesCount": 0,
    "author": {
      "username": "jake",
      "bio": "I work at statefarm",
      "image": "https://i.stack.imgur.com/xHWG8.jpg",
      "following": false
    },
    "averageRating": null,
    "ratingsCount": 0,
  }
}
``` 

`GET api/v1/articles/<str:slug>/ratings` the response should be something like this(or an empty list if the ratingsCount is 0):
```source-json
{
  "ratings": [
    {
      "user": "jake",
      "stars": 5
    },
    {
      "user": "jamie",
      "stars": 3
    }
  ]
}
```

`POST api/v1/articles/<str:slug>/ratings` with the following request body in JSON:
```source-json
{
  "stars": 4
}
```

The response should be something like this:
```source-json
{
  "ratings": {
      "user": "jake",
      "stars": 4
    }
}
```



**What are the relevant pivotal tracker stories?**
[#159053987](https://www.pivotaltracker.com/story/show/159053987)
